easybuild/easyblocks/c/cp2k.py

```diff
@@ -219,22 +230,39 @@ def configure_step(self):
         if elpa:
             options['LIBS'] += ' -lelpa'
             elpa_inc_dir = os.path.join(elpa, 'include', 'elpa-%s' % get_software_version('ELPA'), 'modules')
-            options['FCFLAGSOPT'] += ' -I%s ' % elpa_inc_dir
+            if cp2k_version >= LooseVersion('2024'):
+                options['INCS'] += ' -I%s ' % elpa_inc_dir
+            else:
+                options['FCFLAGSOPT'] += ' -I%s ' % elpa_inc_dir
             if LooseVersion(self.version) >= LooseVersion('6.1'):
                 elpa_ver = ''.join(get_software_version('ELPA').split('.')[:2])
                 options['DFLAGS'] += ' -D__ELPA=%s' % elpa_ver
                 elpa_inc_dir = os.path.join(elpa, 'include', 'elpa-%s' % get_software_version('ELPA'), 'elpa')
-                options['FCFLAGSOPT'] += ' -I%s ' % elpa_inc_dir
+                if cp2k_version >= LooseVersion('2024'):
+                    options['INCS'] += ' -I%s ' % elpa_inc_dir
+                else:
+                    options['FCFLAGSOPT'] += ' -I%s ' % elpa_inc_dir
             else:
                 options['DFLAGS'] += ' -D__ELPA3'
 
         # CUDA
         cuda = get_software_root('CUDA')
         if cuda:
-            options['DFLAGS'] += ' -D__ACC -D__DBCSR_ACC'
-            options['LIBS'] += ' -lcudart -lcublas -lcufft -lrt'
-            options['NVCC'] = ' nvcc'
-
+            if cp2k_version >= LooseVersion('2024'):
+                options['DFLAGS'] += ' -D__OFFLOAD_CUDA -D__DBCSR_ACC '
+                options['LIBS'] += ' -lcufft -lcudart -lnvrtc -lcuda -lcublas'
+                options['OFFLOAD_CC'] = 'nvcc'
```

Comment:
The `OFFLOAD_CC` variable is used in the makefile of CP2K to build files with OpenACC offloading, even versions for older that 2024. For instance 2023.1 fails to build with CUDA support with the old easyblock.

Maybe enable the modification for some versions earlier than 2024 if they are still relevant?

Reply:
Yeah it can surely be modified, but i have not tested them with old version,s when i test them will definitely modify if required. Thanks!!